atlas.go

```diff
@@ -55,8 +77,14 @@ func (n *Node) Path() map[string]string {
 	return result
 }
 
-// AddLink adds the provided key value pair to the tree and returns the new Node that includes all key and values of
-// the parent node and the new key value provided. If a key matches it will have the new value provided here.
+// AddLink adds the key and value strings to the tree
+// and returns the new Node that includes all the key and values of
+// the parent Node plus the new key-value pair provided.
+//
+// If another pair with the same key and a different value already exists in the path
+// then the new Node will have the key replaced with the new provided pair.
+//   (e.g. adding keyA, val2 to a set of {keyA:val1, keyB:val3}
+//   will generate a final path of {keyA: val2, keyB: val3}).
```

Comment:
```suggestion
//   will return set {keyA: val2, keyB: val3}).
```

You are mixing `set` and `path` and using the same ways to talk about them. I find it better to just use the word set in this particular case. But we can also use `path`